README.md

```diff
@@ -411,6 +413,62 @@ When reduced to a prefix list, it would look like this:
 
 Which is then turned into an ste_vec of hashes which can be directly queries against the index.
 
+##### rich_jsonb opts
+
+A rich_jsonb index supports a richer set of JSONB functions and operators than an ste_vec index.
+
+Leaf nodes of the indexed JSONB document are also stored inline in the index which means that subsets of the document (down to individual fields and array elements) can be selected and returned to the client, avoiding the need for the client application to fetch the entire JSONB document.
+
+###### Supported functions & operators
+
+`cs_binop_v1(jsonb, '->', integer)`
+
+Extracts n'th element of JSON array (array elements are indexed from zero, but negative integers count from the end).
+
+`cs_binop_v1(jsonb, '->', text)`
+
+Extracts JSON object field with the given key.
+
+`cs_binop_v1(jsonb, '->>', integer)`
+
+Extracts n'th element of JSON array, as text.
+
+`cs_binop_v1(jsonb, '->>', text)`
```

Comment:
How would `cs_binop_v1` handle the different return types between operators (`jsonb`, `text`, or `bool`)? Can Pl/pgSQL handle multiple return types for a single function?

Reply:
@CDThomas that is a very good question!

Postgres does support limited polymorphism via `anyelement`, `anyarray`, `anynonarray`, `anyenum`, or `anyrange`.

The way it works is that if you have a function `foo(anyelement) -> anyelement` then `anyelement` becomes the same type everywhere it appears.

So `foo(text) -> text` would be allowed, but `foo(text) -> integer` would not.

I think that is sufficient to express all of the variations required by `cs_binop_v1` but I will need to audit to double check.